cfme/containers/provider/openshift.py

```diff
@@ -70,6 +72,37 @@ def __init__(self, name=None, key=None, zone=None,
     def cli(self):
         return OcpCli(self)
 
+    def get_bearer_token(self):
+
+        username = self.endpoints['default'].credentials.principal
+        res = self.cli.run_command('oc login -u={} -p={}'.format(
+            username, self.endpoints['default'].credentials.secret))
+        if not res.success:
+            raise Exception('Failed to login user "{}": "{}"'.format(username, res.output))
+        res = self.cli.run_command('oc whoami -t')
+        if res.success:
+            return res.output.strip()
+        raise Exception('Failed to get Bearer token: "{}"'.format(res.output))
+
+    def get_mgmt_system(self):
+        """ Returns the mgmt_system using the :py:func:`utils.providers.get_mgmt` method.
+        """
+        # gotta stash this in here to prevent circular imports
+        from utils.providers import get_mgmt
+
+        credentials = conf.cfme_data['management_systems'].get(
+            self.key)['endpoints']['default']['credentials']
+        credentials = conf.credentials[credentials]
+        credentials['token'] = self.get_bearer_token()
+
+        if self.key:
+            return get_mgmt(self.key, credentials=credentials)
+        elif hasattr(self, 'provider_data'):
+            return get_mgmt(self.provider_data, credentials=credentials)
+        else:
+            raise ProviderHasNoKey(
+                'Provider {} has no key, so cannot get mgmt system'.format(self.name))
+
```

Comment:
Can you please explain what is going on here? You can use `prov.endpoints['default'].token` to get token and `prov.mgmt` to get management system. Why do we need these two methods here?